### PR TITLE
Compressor: Add data format(QZ_DEFLATE_GZIP_EXT) for QAT Zlib

### PR DESCRIPTION
**Pull request by hualongfeng**
QAT zlib 'QZ_DEFLATE_RAW' data format cannot decompress by QAT hardware. So there is 'QZ_DEFLATE_GZIP_EXT' data
format added.
    
'QZ_DEFLATE_GZIP_EXT' data format need to add gz_header by deflateSetHeader() in QATzip. And it may lead multi stream
in one compression because for hardware buffer. So the windows bit is important information for decompression, which related to
if the inflate remove header.
    
Add busy_polling setting for reducing latency when busy_polling is true

When I upload a object to RGW by multipart, the head object xattr(user.rgw.compression) don't have compressor_message
when the value should be valid and part object xattr(user.rgw.compression) have compressor_message. 

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by mkogan1** [comment]
@hualongfeng when running with `qat_compressor_data_fmt=QZ_DEFLATE_GZIP_EXT`
there is a problem that is observed with `s3cmd get` - the length of the data is less than that of the correct file size:

```
vstart.sh ... -o qat_compressor_enabled=true --rgw_compression zlib  -o qat_compressor_data_fmt=QZ_DEFLATE_GZIP_EXT


❯ ls -l CentOS-Stream-9-latest-x86_64-dvd1.iso
-rw-r--r--. 1 ceph ceph 10824122368 Jun 17 11:03 CentOS-Stream-9-latest-x86_64-dvd1.iso
                        ^^^^^^^^^^^

❯ s3cmd put --host=0:8000 --multipart-chunk-size-mb=150 ./CentOS-Stream-9-latest-x86_64-dvd1.iso s3://bkt/objMP01
                        
❯ s3cmd get --host=0:8000 s3://bkt/objMP01 --force
download: 's3://bkt/objMP01' -> './objMP01'  [1 of 1]
 10344660992 of 10824122368	95% in   22s   448.17 MB/s <- stuck
 ^x^x^x^x^x^    ^^^^^^^^^^^  

❯ tail -F ./out/radosgw.8000.log
2024-06-19T08:34:18.799-0700 7fffe5be3640  1 beast: 0x7fffbc38f720: 127.0.0.1 - cosbench
 [19/Jun/2024:08:33:55.810 -0700] "GET /bkt/objMP01 HTTP/1.1" 200 10820452352 - - - latency=22.988271713s
                                                                  ^X^X^X^X^X^             
```

**Comment by hualongfeng** [comment]
> @hualongfeng when running with `qat_compressor_data_fmt=QZ_DEFLATE_GZIP_EXT` there is a problem that is observed with `s3cmd get` - the length of the data is less than that of the correct file size:
> 
> ```
> vstart.sh ... -o qat_compressor_enabled=true --rgw_compression zlib  -o qat_compressor_data_fmt=QZ_DEFLATE_GZIP_EXT
> 
> 
> ❯ ls -l CentOS-Stream-9-latest-x86_64-dvd1.iso
> -rw-r--r--. 1 ceph ceph 10824122368 Jun 17 11:03 CentOS-Stream-9-latest-x86_64-dvd1.iso
>                         ^^^^^^^^^^^
> 
> ❯ s3cmd put --host=0:8000 --multipart-chunk-size-mb=150 ./CentOS-Stream-9-latest-x86_64-dvd1.iso s3://bkt/objMP01
>                         
> ❯ s3cmd get --host=0:8000 s3://bkt/objMP01 --force
> download: 's3://bkt/objMP01' -> './objMP01'  [1 of 1]
>  10344660992 of 10824122368	95% in   22s   448.17 MB/s <- stuck
>  ^x^x^x^x^x^    ^^^^^^^^^^^  
> 
> ❯ tail -F ./out/radosgw.8000.log
> 2024-06-19T08:34:18.799-0700 7fffe5be3640  1 beast: 0x7fffbc38f720: 127.0.0.1 - cosbench
>  [19/Jun/2024:08:33:55.810 -0700] "GET /bkt/objMP01 HTTP/1.1" 200 10820452352 - - - latency=22.988271713s
>                                                                   ^X^X^X^X^X^             
> ```

Hi Mark  I tested your case. I found one block compression ratio is huge. the size  1048576B object compressed to size 7477B object. And one compression is part data to return. So it is less than that of the correct file size. I have fixed the error.

**Comment by mkogan1** [comment]
@hualongfeng thank you, after the rebase with QZ_DEFLATE_GZIP_EXT the GET is working.

regarding
> QZ_DEFALTE_RAW data format is in raw DEFALTE without any additional header(This data format only support compression, decompression will fallback to software).
testing with QZ_DEFALTE_RAW when the PUT is via QAT there is an error when performing a instead of falling back to software:

```
❯ s3cmd get --host=0:8001 s3://bkt/objMP01 --force
download: 's3://bkt/objMP01' -> './objMP01'  [1 of 1]
ERROR: Download of './objMP01' failed (Reason: 403 (AccessDenied))
ERROR: S3 error: 403 (AccessDenied)

less ./out/radosgw.8001.log
...
│2024-06-24T04:19:44.735-0700 7fffe8be9640 10 Compression for rgw is enabled, decompress part bl_ofs=0, bl_len=4194304                                                                      
│2024-06-24T04:19:44.735-0700 7fffe8be9640  1 ZlibCompressor: Decompression error: decompress return -3                                                                                     
│2024-06-24T04:19:44.735-0700 7fffe8be9640 -1 Decompression failed with exit code -1                                                                                                        
│2024-06-24T04:19:44.735-0700 7fffe8be9640  0 req 9157827340147204404 0.010000112s s3:get_obj iterate_obj() failed with -1                                                                  
│2024-06-24T04:19:44.743-0700 7fffe33de640  2 req 9157827340147204404 0.018000200s s3:get_obj completing                                                                                    
│2024-06-24T04:19:44.743-0700 7fffe33de640 10 req 9157827340147204404 0.018000200s cache get: name=default.rgw.log++script.postrequest. : hit (negative entry)                              
│2024-06-24T04:19:44.743-0700 7fffe33de640  2 req 9157827340147204404 0.018000200s s3:get_obj op status=-1                                                                                  
│2024-06-24T04:19:44.743-0700 7fffe33de640  2 req 9157827340147204404 0.018000200s s3:get_obj http status=403                                                                               
│2024-06-24T04:19:44.743-0700 7fffe33de640  1 ====== req done req=0x7fffbc38f720 op status=-1 http_status=403 latency=0.018000200s ======
```

**Comment by hualongfeng** [comment]
> @hualongfeng thank you, after the rebase with QZ_DEFLATE_GZIP_EXT the GET is working.
> 
> regarding
> 
> > QZ_DEFALTE_RAW data format is in raw DEFALTE without any additional header(This data format only support compression, decompression will fallback to software).
> > testing with QZ_DEFALTE_RAW when the PUT is via QAT there is an error when performing a instead of falling back to software:
> 
> ```
> ❯ s3cmd get --host=0:8001 s3://bkt/objMP01 --force
> download: 's3://bkt/objMP01' -> './objMP01'  [1 of 1]
> ERROR: Download of './objMP01' failed (Reason: 403 (AccessDenied))
> ERROR: S3 error: 403 (AccessDenied)
> 
> less ./out/radosgw.8001.log
> ...
> │2024-06-24T04:19:44.735-0700 7fffe8be9640 10 Compression for rgw is enabled, decompress part bl_ofs=0, bl_len=4194304                                                                      
> │2024-06-24T04:19:44.735-0700 7fffe8be9640  1 ZlibCompressor: Decompression error: decompress return -3                                                                                     
> │2024-06-24T04:19:44.735-0700 7fffe8be9640 -1 Decompression failed with exit code -1                                                                                                        
> │2024-06-24T04:19:44.735-0700 7fffe8be9640  0 req 9157827340147204404 0.010000112s s3:get_obj iterate_obj() failed with -1                                                                  
> │2024-06-24T04:19:44.743-0700 7fffe33de640  2 req 9157827340147204404 0.018000200s s3:get_obj completing                                                                                    
> │2024-06-24T04:19:44.743-0700 7fffe33de640 10 req 9157827340147204404 0.018000200s cache get: name=default.rgw.log++script.postrequest. : hit (negative entry)                              
> │2024-06-24T04:19:44.743-0700 7fffe33de640  2 req 9157827340147204404 0.018000200s s3:get_obj op status=-1                                                                                  
> │2024-06-24T04:19:44.743-0700 7fffe33de640  2 req 9157827340147204404 0.018000200s s3:get_obj http status=403                                                                               
> │2024-06-24T04:19:44.743-0700 7fffe33de640  1 ====== req done req=0x7fffbc38f720 op status=-1 http_status=403 latency=0.018000200s ======
> ```

Hi @mkogan1 Sorry, it is my typo error.  It should be "QZ_DEFLATE_RAW", which work. I'll fix the type error.

**Comment by mkogan1** [comment]
@hualongfeng thank you the fix for 
> Hi @mkogan1 Sorry, it is my typo error. It should be "QZ_DEFLATE_RAW", which work. I have fixed the typo error.

is working

**Comment by mkogan1** [comment]
@hualongfeng 
when checking an upgrade scenario, several permutations exhibit QAT errors,
your advice, please

image

1. on a vstart where the _CentOS-Stream-9-latest-x86_64-dvd1.iso_ was s3cmd **PUT** when `qat_compressor_enabled=false` or `qat_compressor_enabled=true` 
when performing s3cmd **GET** with this PR and _DEFLATE_GZIP_EXT_ there is a `QatAccel: QAT compressor NOT OK` error (instead of fallback to software ?)
```
│2024-06-26T01:30:41.250-0700 7fffe63e4640  1 ====== starting new request req=0x7fffbc38f720 =====                                                                                          │
│2024-06-26T01:30:47.460-0700 7fffe83e8640  1 QatAccel: QAT compressor NOT OK                                                                                                               │
│2024-06-26T01:30:47.460-0700 7fffe83e8640 -1 Decompression failed with exit code -1                                                                                                        │
│2024-06-26T01:30:47.460-0700 7fffe83e8640  0 req 9716899533513771753 6.210071564s s3:get_obj iterate_obj() failed with -1                                                                  │
│2024-06-26T01:30:47.462-0700 7fffe83e8640  1 ====== req done req=0x7fffbc38f720 op status=-1 http_status=403 latency=6.212071419s ======                                                   │
│2024-06-26T01:30:47.462-0700 7fffe83e8640  1 beast: 0x7fffbc38f720: 127.0.0.1 - cosbench [26/Jun/2024:01:30:41.250 -0700] "GET /bkt/objMP01 HTTP/1.1" 403 232 - - - latency=6.212071419s 
```

2. on a vstart where the _CentOS-Stream-9-latest-x86_64-dvd1.iso_ was s3cmd **PUT** with this PR and _DEFLATE_GZIP_EXT_, 
when performing s3cmd **GET** with this PR and _QZ_DEFLATE_RAW_ there is a `QatAccel: QAT compressor DATA ERROR` error (instead of fallback to software ?)
```
│2024-06-26T08:22:25.819-0700 7fffe5be3640  1 ====== starting new request req=0x7fffbc38f720 =====                                                                                          │
│2024-06-26T08:22:32.035-0700 7fffe7be7640  1 QatAccel: QAT compressor DATA ERROR                                                                                                           │
│2024-06-26T08:22:32.035-0700 7fffe7be7640 -1 Decompression failed with exit code -1                                                                                                        │
│2024-06-26T08:22:32.035-0700 7fffe7be7640  0 req 11210452492432157814 6.216071606s s3:get_obj iterate_obj() failed with -1                                                                 │
│2024-06-26T08:22:32.037-0700 7fffe7be7640  1 ====== req done req=0x7fffbc38f720 op status=-1 http_status=403 latency=6.218071461s ======                                                   │
│2024-06-26T08:22:32.037-0700 7fffe7be7640  1 beast: 0x7fffbc38f720: 127.0.0.1 - cosbench [26/Jun/2024:08:22:25.819 -0700] "GET /bkt/objMP01 HTTP/1.1" 403 232 - - - latency=6.218071461s
```

**Comment by hualongfeng** [comment]
hi @mkogan1 
> 1. on a vstart where the _CentOS-Stream-9-latest-x86_64-dvd1.iso_ was s3cmd **PUT** when `qat_compressor_enabled=false` or `qat_compressor_enabled=true`
>    when performing s3cmd **GET** with this PR and _DEFLATE_GZIP_EXT_ there is a `QatAccel: QAT compressor NOT OK` error (instead of fallback to software ?)
> 

This error has not been reproduced here. Can you re-verify it with the updated PR?
（Operation: The PUT operation is performed under the condition of qat_compressor_enabled=false, and then ceph.conf is changed to qat_compressor_enabled=true, qat_compressor_data_fmt=QZ_DEFLATE_GZIP_EXT, RGW is restarted, and then the GET operation is performed.）

> 
> 2. on a vstart where the _CentOS-Stream-9-latest-x86_64-dvd1.iso_ was s3cmd **PUT** with this PR and _DEFLATE_GZIP_EXT_,
>    when performing s3cmd **GET** with this PR and _QZ_DEFLATE_RAW_ there is a `QatAccel: QAT compressor DATA ERROR` error (instead of fallback to software ?)
> 

This error is because once the QAT compression/decompression side is initialized, the data format will not be changed. I have now restricted QAT decompression. Only when data_fmt=QZ_DEFLATE_GZIP_EXT and the data is also in this format, will it be adjusted to QAT decompression.

**Comment by mkogan1** [comment]
@hualongfeng thank you very much,
the `QatAccel: QAT compressor DATA ERROR` has been resolved
image

regarding the `QatAccel: QAT compressor NOT OK` the flow to reproduce is like performing an upgrade scenario, can be reproduced in several ways, will describe below one possible flow in detail:

1. git branch  -> hualongfeng:wip_qat_zlib_data_format
   git reset --hard ed6f4b0696d11ef8c3694e0e6156a97b025b3786 
   (before the 2 PR commits)
   ninja
2. vstart.sh ... -o qat_compressor_enabled=false
   The PUT operation is performed under the condition of
   qat_compressor_enabled=false
3. pkill radosgw  
   git pull 
   ninja
4. and then ceph.conf is changed to qat_compressor_enabled=true,   
   qat_compressor_data_fmt=QZ_DEFLATE_GZIP_EXT, 
   RGW is restarted, and then the GET operation is performed.


question please, what are the pros/cons of leaving `QZ_DEFLATE_RAW` as the default?

**Comment by mkogan1** [comment]
jenkins test make check

**Comment by mkogan1** [comment]
jenkins test make check

**Comment by mkogan1** [comment]
@hualongfeng 
to facilitate backward compatibility for users that already have data written with gzip or QZ_DEFLATE_RAW could we add a backward compatibility flag that will explicitly mention that data which will be written with QZ_DEFLATE_GZIP_EXT will not be able to be read when the qat_decompressor_data_fmt_raw=true is enabled

like possible example:

in src/common/options/global.yaml.in
```
- name: qat_decompressor_data_fmt_raw
  type: bool
  level: advanced
  desc: Enable RAW GZIP data format within Qatzip when using QAT dompressor
  long_desc: Data that will be written with QZ_DEFLATE_GZIP_EXT will not be able to be read when the qat_decompressor_data_fmt_raw=true is enabled (A QAT error would occur)
  default: false
...
in src/compressor/zlib/ZlibCompressor.cc
```
if (qat_enabled && (!compressor_message || *compressor_message == GZIP_WRAPPER + MAX_WBITS ) && (g_ceph_context->_conf.get_val<std::string>("qat_compressor_data_fmt").compare("QZ_DEFLATE_GZIP_EXT") == 0 || g_ceph_context->_conf.get_val<bool>("qat_decompressor_data_fmt_raw")))
```

**Comment by hualongfeng** [comment]
> question please, what are the pros/cons of leaving `QZ_DEFLATE_RAW` as the default?

@mkogan1  I still recommend using QZ_DEFLATE_GZIP_EXT as the default value, so that QAT can be used for both compression and decompression when QAT is available. If you use QZ_DEFLATE_RAW as the default, then decompression will not work with QAT without reconfiguring. 

The previous switch to QAT decompression was due to a mistake in the judgment logic. Now the above two phenomena have been solved.

**Comment by hualongfeng** [comment]
> @hualongfeng to facilitate backward compatibility for users that already have data written with gzip or QZ_DEFLATE_RAW could we add a backward compatibility flag that will explicitly mention that data which will be written with QZ_DEFLATE_GZIP_EXT will not be able to be read when the qat_decompressor_data_fmt_raw=true is enabled
> 
> like possible example:
> 
> in src/common/options/global.yaml.in
> 
> ```
> - name: qat_decompressor_data_fmt_raw
>   type: bool
>   level: advanced
>   desc: Enable RAW GZIP data format within Qatzip when using QAT dompressor
>   long_desc: Data that will be written with QZ_DEFLATE_GZIP_EXT will not be able to be read when the qat_decompressor_data_fmt_raw=true is enabled (A QAT error would occur)
>   default: false
> ```
> 
> in src/compressor/zlib/ZlibCompressor.cc
> 
> ```
> if (qat_enabled && (!compressor_message || *compressor_message == GZIP_WRAPPER + MAX_WBITS ) &&
>    (g_ceph_context->_conf.get_val<std::string>("qat_compressor_data_fmt").compare("QZ_DEFLATE_GZIP_EXT") == 0 || g_ceph_context->_conf.get_val<bool>("qat_decompressor_data_fmt_raw")))
> ```

Hi @mkogan1 `(!compressor_message || *compressor_message == GZIP_WRAPPER + MAX_WBITS ) ` changed to `(compressor_message.has_value() && *compressor_message == GZIP_WRAPPER + MAX_WBITS )` is enough.  It has been changed.  
The value of compressor_message before this PR can only be -15 or empty. After this judgment, it will not turn to QAT decompression.

**Comment by mkogan1** [comment]
@hualongfeng thank you for explaining, below is an updated green-field (new installation) and brown-field (upgrade) QAT inter-op scenario

image

as it stands - there are 2 items/permutations that think still worth a mention with context to previous discussions

1. currently the only way to get QAT HW Decode to operate is only with one permutation when both PUT and GET were/are performed with `QZ_DEFLATE_GZIP_EXT`.
2. the `QZ_DEFLATE_RAW` currently can only performs QAT HW encoding.

@hualongfeng @cbodley are we satisfied with the current permutations state

**Comment by hualongfeng** [comment]
> @hualongfeng @cbodley are we satisfied with the current permutations state
Hi @mkogan1  I am satisfied with the current permutations state. 
Hi @cbodley Do you have any advice?

**Comment by hualongfeng** [comment]
jenkins test make check

**Review by cbodley — COMMENTED** [review]
is there no way to get hardware acceleration when decompressing objects written with non-qat zlib? that's unfortunate for users with lots of already-compressed data. is there any advice we can offer there? i don't know that rgw can re-compress the data in place

this behavior is probably worth documenting in https://docs.ceph.com/en/latest/radosgw/qat-accel/ or https://docs.ceph.com/en/latest/radosgw/compression/, along with any new config options

**Comment by cbodley** [comment]
we'll want this fixed for the squid release too, so i created https://tracker.ceph.com/issues/66878 to track the backport

**Comment by cbodley** [comment]
i also noticed that https://docs.ceph.com/en/latest/radosgw/qat-accel/ wasn't updated in https://github.com/ceph/ceph/pull/54624, so opened https://tracker.ceph.com/issues/66879 to track that

**Comment by hualongfeng** [comment]
> is there no way to get hardware acceleration when decompressing objects written with non-qat zlib? that's unfortunate for users with lots of already-compressed data. is there any advice we can offer there? i don't know that rgw can re-compress the data in place
> 

For objects whose current compression format is QZ_DEFLATE_RAW format, there is no way to obtain hardware acceleration. But we can submit a new PR to change the software compression data format to QZ_DEFLATE_GZIP or QZ_DEFLATE_GZIP_EXT format. In this way, non-qat zlib compressed objects can also obtain hardware acceleration when decompressing. However, this way cannot take effect on existing zlib objects.

**Comment by hualongfeng** [comment]
> i also noticed that https://docs.ceph.com/en/latest/radosgw/qat-accel/ wasn't updated in #54624, so opened https://tracker.ceph.com/issues/66879 to track that

https://github.com/ceph/ceph/pull/58641   doc update has submitted.

**Comment by hualongfeng** [comment]
> we'll want this fixed for the squid release too, so i created https://tracker.ceph.com/issues/66878 to track the backport

https://github.com/ceph/ceph/pull/58643   backport action done.

**Comment by hualongfeng** [comment]
> > For objects whose current compression format is QZ_DEFLATE_RAW format, there is no way to obtain hardware acceleration. But we can submit a new PR to change the software compression data format to QZ_DEFLATE_GZIP or QZ_DEFLATE_GZIP_EXT format.
> 
> this refers to the software compression done by `QatAccel`, right? i'm asking about objects compressed by `ZlibCompressor` when `qat_compressor_enabled=false`

Hi @cbodley  software compression refers to `ZlibCompressor`. In other words, if we want the data compressed by `ZlibCompressor` to be decompressed through QAT in the future, we need to change the data format of `ZlibCompressor` to `QZ_DEFLATE_GZIP` or `QZ_DEFLATE_GZIP_EXT`. In this way, the new data is compressed by `ZlibCompressor` and can also be decompressed by QAT.

File: src/common/options/global.yaml.in [hunk 1]
```diff
@@ -790,6 +790,17 @@ options:
   level: advanced
   desc: Set the maximum number of session within Qatzip when using QAT compressor
   default: 256
+- name: qat_compressor_data_fmt
```
[the remark this answers is not in the document]
Comment (cbodley) [reply src/common/options/global.yaml.in:793]: @hualongfeng ^ can this option be removed?

[the remark this answers is not in the document]
Comment (hualongfeng) [reply src/common/options/global.yaml.in:793]: > @hualongfeng ^ can this option be removed?

Hi @cbodley  Yes, This option can be removed.  This parameter can be achieved by changing other parameters.

[the remark this answers is not in the document]
Comment (hualongfeng) [reply src/common/options/global.yaml.in]: @cbodley  Removed.

**Review by cbodley — APPROVED** [review]

**Review by mkogan1 — APPROVED** [review]
Updated test matrix post last commit which removes `QZ_DEFLATE_RAW`
image
the only notable item to keep in mind is that the RGW before this PR will not be able to read **QAT** compressed (_ONLY compressed by QAT HW_) objects written after this PR

**Comment by cbodley** [comment]
> the only notable item to keep in mind is that the RGW before this PR will not be able to read **QAT** compressed (_ONLY compressed by QAT HW_) objects written after this PR

thanks, then we'll need to make sure this lands in time for the squid release

**Comment by cbodley** [comment]
jenkins test api

**Comment by cbodley** [comment]
jenkins test make check

**Comment by cbodley** [comment]
passed rgw suite in https://pulpito.ceph.com/cbodley-2024-07-26_01:21:30-rgw-wip-cbodley-testing-distro-default-smithi/

**Pull request merged**